src/components/toolbar/Toolbar.tsx

```diff
@@ -185,32 +178,28 @@ function ToolbarItemInternal(props: ToolbarItemInternalProps) {
         width: 'fit-content',
       }}
       icon={
-        <div
-          style={{
-            display: 'flex',
-            justifyContent: 'center',
-            alignItems: 'center',
-            width: 0,
-            height: 0,
-            marginRight: 0,
-          }}
-        >
-          <Icon icon={resizedIcon} size={16} />
-          {isPopover && (
-            <Icon
-              icon="caret-right"
-              size={9}
-              style={{
-                transform: 'rotate(45deg)',
-                position: 'absolute',
-                bottom: 0,
-                right: 0,
-                width: 9,
-                height: 9,
-              }}
-            />
+        <>
+          {isPopover ? (
+            <>
+              {/*For the icon to be properly styled, it should not have any siblings*/}
+              <div style={{ display: 'contents' }}>{normalizedIcon}</div>
+              <Icon
+                icon="caret-right"
+                size={9}
+                style={{
+                  transform: 'rotate(45deg)',
+                  position: 'absolute',
+                  bottom: 0,
+                  right: 0,
+                  width: 9,
+                  height: 9,
+                }}
+              />
```

Comment:
Do this is possible to create the Icon with styled ?

Reply:
It's easier not to use styled here because the width and height need to override the width and height set by the `bp5-icon` class.